### PR TITLE
Attempted fix to hyperoptimization

### PR DESCRIPTION
**Pull request by kevingreenman**
@swansonk14 This passed all the tests except the RDKit one (I think because I forked before you made the change in response to Florence's message). I essentially used `num_iters` as a flag of whether training or hyperoptimization was happening (None for training, int for hyperopt) and duplicated what run_training.py was doing on the test set for the validation set. I tried to do something that was minimally disruptive to the code, but there might be a more elegant way to do it. This might also have some effect on another piece of the code that I'm not foreseeing.

### CODE REVIEW

**Comment by cjmcgill** [comment]
There may be a way to do this and keep it contained within the hyperopt files instead of affecting the core training files. You can have the hyperopt code feed certain args to the individual calculations. So maybe you have it feed a seed number the same as what you'd use in training so that it splits your data for hyperoptimization the same way as in training.

**Comment by kevingreenman** [comment]
Based on discussions in the Green group, I think the current method of doing hyperparameter optimization is okay so long as users read the warning note in the README.
